idaes/models/properties/modular_properties/phase_equil/smooth_VLE.py

```diff
@@ -50,116 +50,288 @@ def phase_equil(b, phase_pair):
-
-        # PSE paper Eqn 13
-        def rule_t1(b):
-            return _t1 == smooth_max(
-                b.temperature, b.temperature_bubble[phase_pair], eps_1
-            )
+        # Equilibrium temperature
+        def rule_tbar(b):
+            return b.tbar[phase_pair] - b.temperature - \
+                s['Vap'] + s['Liq'] == 0
+        b.add_component("tbar_constraint" + suffix, Constraint(rule=rule_tbar))
 
-        b.add_component("_t1_constraint" + suffix, Constraint(rule=rule_t1))
     else:
-        _t1 = b.temperature
+        b.tbar[phase_pair] = b.temperature
 
-    if l_only_comps == []:
-        b.add_component(
-            "eps_2" + suffix,
-            Param(
-                default=0.0005,
-                mutable=True,
-                doc="Smoothing parameter for Teq",
-                units=t_units,
-            ),
-        )
-        eps_2 = getattr(b, "eps_2" + suffix)
+    b.add_component(
+        "eps" + suffix,
+        Param(default=1e-04,
+              mutable=True,
+              doc='Smoothing parameter for complementarities',
+              units=t_units),
+    )
+    eps = getattr(b, "eps" + suffix)
+    
+    b.add_component(
+        "gp" + suffix,
+        Var(b.params.phase_list,
+            initialize=0.0,
+            bounds=(0, None),
+            doc='Slack variable for cubic second derivative for phase p',
+            units=pyunits.dimensionless),
+    )
+    gp = getattr(b, "gp" + suffix)
+    
+    b.add_component(
+        "gn" + suffix,
+        Var(b.params.phase_list,
+            initialize=0.0,
+            bounds=(0, None),
+            doc='Slack variable for cubic second derivative for phase p',
+            units=pyunits.dimensionless),
+    )
+    gn = getattr(b, "gn" + suffix)
 
-        # PSE paper Eqn 14
-        # TODO : Add option for supercritical extension
-        def rule_teq(b):
-            return b._teq[phase_pair] == smooth_min(
-                _t1, b.temperature_dew[phase_pair], eps_2
-            )
-
-    elif v_only_comps == []:
+    def rule_temperature_slack_complementarity(b, p):
+        flow_phase = b.flow_mol_phase[p]
+        if b.params.config.supercritical:
+            if p == 'Vap':
+                return smooth_min(s[p], flow_phase, eps) == 0
+            else:
+                return Constraint.Skip
+        else:
+            return smooth_min(s[p], flow_phase, eps) == 0
+    b.add_component(
+        "temperature_slack_complementarity" + suffix,
+        Constraint(
+            b.params.phase_list,
+            rule=rule_temperature_slack_complementarity,
+        ),
+    )
 
-        def rule_teq(b):
-            return b._teq[phase_pair] == _t1
+    def rule_cubic_root_complementarity(b, p):
+        p1, p2 = phase_pair
+        return b.cubic_second_derivative[p1, p2, p] == gp[p] - gn[p]
+    b.add_component(
+        "cubic_root_complementarity" + suffix,
+        Constraint(b.params.phase_list, rule=rule_cubic_root_complementarity),
+    )
```

Comment:
If we're directly using things referencing the cubic equation of state, I think this should be made a separate phase equilibrium method. As it stands right now, we can do phase equilibria with the ideal  EoS because methods for calculating the bubble and dew points are implemented either from the Antoine equation or Henry's Law (although the Henry's Law implementation is broken at the moment, see #718).